keps/sig-node/4576-conditions-for-pod-terminated/README.md

```diff
@@ -0,0 +1,906 @@
+-->
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+<!--
+This section is incredibly important for producing high-quality, user-focused
+documentation such as release notes or a development roadmap. It should be
+possible to collect this information before implementation begins, in order to
+avoid requiring implementors to split their attention between writing release
+notes and implementing the feature itself. KEP editors and SIG Docs
+should help to ensure that the tone and content of the `Summary` section is
+useful for a wide audience.
+
+A good summary is probably at least a paragraph in length.
+
+Both in this section and below, follow the guidelines of the [documentation
+style guide]. In particular, wrap lines to a reasonable length, to make it
+easier for reviewers to cite specific portions, and to minimize diff churn on
+updates.
+
+[documentation style guide]: https://github.com/kubernetes/community/blob/master/contributors/guide/style-guide.md
+-->
+This KEP proposes a
+`PodTerminated` condition in pod status to indicate a pod has
+reached a state where it's containers have terminated.
+
+## Motivation
+
+<!--
+This section is for explicitly listing the motivation, goals, and non-goals of
+this KEP.  Describe why the change is important and the benefits to users. The
+motivation section can optionally provide links to [experience reports] to
+demonstrate the interest in a KEP within the wider Kubernetes community.
+
+[experience reports]: https://github.com/golang/go/wiki/ExperienceReports
+-->
+
+There are two phases, Succeeded and Failed, which are used to indicate that 
+all containers in the pod have been terminated.  
+When all containers of the pod are terminated, the kubelet will update the phase of 
+the pod to Succeeded or Failed in the SyncPod,  in which case it is not safe to remove the corresponding resource dependencies 
+such as pvc protection, because the phase of the pod may already be in the Succeeded or Failed state 
+but the volumes are not unmounted(the umount process is asynchronous with the pod's phase change).
+
+A new dedicated condition `PodTerminated` indicate that
+all containers in the pod have been terminated and corresponding resource dependencies have been processed
+- volumes are unmounted
+- secrets are unregistered
+- configmaps are unregistered
+- cgroups are removed
+- userns are released
+
+
+### Goals
+
+<!--
+List the specific goals of the KEP. What is it trying to achieve? How will we
+know that this has succeeded?
+--> 
+Custom pod controllers/operators can use a dedicated condition `PodTerminated` indicating
+a pod has
+reached a state where it's containers have terminated and corresponding resource dependencies have been processed.
+
```

Comment:
Should we consider solving the that created this in this KEP also?

Reply:
> Should we consider solving the that created this in this KEP also?

Do you meaning this issue https://github.com/kubernetes/kubernetes/issues/120756?

Reply:
Yes.